python/PiFinder/ui/wifi_password.py

```diff
@@ -0,0 +1,297 @@
+import logging
+import qrcode
+import math
+
+from PiFinder import state_utils
+from PiFinder.ui.base import UIModule
+from PiFinder.utils import get_sys_utils
+from PiFinder.ui.marking_menus import MarkingMenuOption, MarkingMenu
+# from PiFinder.sys_utils import Network
+
+logger = logging.getLogger("WiFiPassword")
+
+# Constants for Display Modes
+DM_QR = 0  # Display QR code for scanning with smartphone or tablet.
+DM_PLAIN_PWD = 1  # Display plain password
+DM_CLIENT = 2
+DM_LAST = DM_PLAIN_PWD
+
+
+class UIWiFiPassword(UIModule):
+    """
+    UI for displaying the Access Point name and password.
+    """
+
+    __help_name__ = "wifi_password"
+    __title__ = "WIFI"
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.network = get_sys_utils().Network()
+        self._update_info()
+
+        self.qr_image = None
+
+        if self.ap_mode == "Client":
+            self.wifi_display_mode = DM_CLIENT
+            self.marking_menu = MarkingMenu(
+                left=MarkingMenuOption(),
+                right=MarkingMenuOption(),
+                down=MarkingMenuOption(),
+            )
+        else:
+            # Default to QR code display
+            self.wifi_display_mode = DM_QR
+            # self.wifi_display_mode = DM_PLAIN_PWD
+
+            self.marking_menu = MarkingMenu(
+                left=MarkingMenuOption(
+                    label="QR", callback=self.mm_display_qr, enabled=True
+                ),
+                right=MarkingMenuOption(
+                    label="Passwd", callback=self.mm_display_pwd, enabled=True
+                ),
+                down=MarkingMenuOption(
+                    label="Exit", callback=self.mm_switch_mode, enabled=True
```

Comment:
There is a built in marking menu system to make this easier:

`MarkingMenuOption(label="SwitchMode", menu_jump="wifi_mode")`